src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/PermanentStorage.java

```diff
@@ -25,4 +26,6 @@
 	 */
 	boolean delete(SapsImage task) throws PermanentStorageException;
 
+	List<AccessLink> generateAccessLinks(String taskId) throws PermanentStorageException;
+
```

Comment:
why the two older methods (methods, not function ok?) receive a SapsImage (this name is odd) and this new one a taskId?

Reply:
This question was mentioned [here](https://github.com/ufcg-lsd/saps-engine/pull/80#discussion_r377227309). At the moment, I don't see any use in passing SapsImage object except its id, but it may be that in some implementation need some other information about the object, what do you suggest?

Reply:
I'd suggest to use `taskId` everywhere